lib/action_subscriber/synchronizer.rb

```diff
@@ -0,0 +1,15 @@
+require "thread"
+module ActionSubscriber
+  class Synchronizer
+    def initialize(delegate)
+      @delegate = delegate
+      @mutex = ::Thread::Mutex.new
```

Comment:
That must be a new thing. It looks like the top-level `Mutex` is documented in both [2.2.0](https://ruby-doc.org/core-2.2.0/Mutex.html) and in [1.9.3](https://ruby-doc.org/core-1.9.3/Mutex.html) so I'll use that instead.